ITKImageProcessingFilters/IlluminationCorrection.cpp

```diff
@@ -198,8 +198,7 @@ void calculateOutputValues(IlluminationCorrection* filter)
 
   typename AccumDataArrayType::Pointer accumulateArrayPtr = AccumDataArrayType::CreateArray(backgroundArrayPtr->getNumberOfTuples(), "Accumulation Array", true);
   accumulateArrayPtr->initializeWithZeros();
-  DataArray<AccumType>& accumArray = *accumulateArrayPtr;
```

Comment:
Put this line back in and use accumArray again. It improves readability and removes the need to constantly dereference the matching smart pointer.

If you can always use `->` then the readability will not be affected, but the use of square brackets prevents that. In the future, if you're calling a method, you can use `->` instead of dereferencing first and then calling the method. For this case, since you'd have to mix and match, just use accumArray.